src/game/enemy/systems.rs

```diff
@@ -288,32 +290,44 @@ pub fn catch_shot_event(
     }
 }
 
-// TODO: How should we determine when an enemy shoots the player?
 /// Enemy changes sprite sheet to shooting and triggers shot event
 pub fn enemy_shoot_player(
     mut commands: Commands,
     mut enemy_q: Query<(Entity, &Transform), (With<Enemy>, Without<Firing>)>,
     enemy_handles: Res<EnemyHandles>,
     mut enemy_shot_player_event_writer: EventWriter<PlayerShotEvent>,
-    keyboard_input: Res<Input<KeyCode>>, // Remove later
+    player_q: Query<&Transform, With<Player>>,
+    win_q: Query<&Window, With<PrimaryWindow>>,
 ) {
-    if keyboard_input.just_pressed(KeyCode::F2) {
-        println!("<< shoot >>");
-        let mut rng = rand::thread_rng();
-        if let Some((enemy_entity, transform)) = enemy_q.iter_mut().choose(&mut rng) {
-            // inserting this replaces the old one
-            commands.entity(enemy_entity).insert((
-                SpriteSheetBundle {
-                    texture_atlas: enemy_handles.soviet_fire.clone(),
-                    sprite: TextureAtlasSprite::new(0),
-                    transform: *transform,
-                    ..default()
-                },
-                EnemyAnimations::SovietFire.get_indices(),
-                EnemyAnimations::SovietFire.get_timer(),
-                Firing::default(),
-            ));
-            enemy_shot_player_event_writer.send(PlayerShotEvent);
+    let win = win_q.get_single().unwrap();
+    if let Ok(player_transform) = player_q.get_single() {
+        for (enemy_entity, enemy_transform) in enemy_q.iter_mut() {
+            // If enemy is on screen, theres a chance to shoot player
+            let distance = player_transform
+                .translation
+                .distance(enemy_transform.translation)
+                + 0.1;
+            if distance < f32::max(win.height(), win.width()) {
+                let shot_chance = distance * random::<f32>();
```

Comment:
This somewhat works. I want it to scale with distance in some way and this seems to on average